File: core/ReverseOnlyLetters/ReverseOnlyLetters.py
def reverseOnlyLetters(s: str) -> str:
    left, right = 0, len(s) - 1
    res = list(s)
    while left < right:
        while left < right and not res[left].isalpha():
            left += 1
        while left < right and not res[right].isalpha():
            right -= 1
        temp = res[left]
        res[left] = res[right]
        res[right] = temp

        left += 1
        right -= 1

    return "".join(res)


if __name__ == '__main__':
    print(reverseOnlyLetters("a-bC-dEf-ghIj"))

File: core/ReverseOnlyLetters/ReverseOnlyLetters.java
package com.dbc;

public class ReverseOnlyLetters {
    public String reverseOnlyLetters(String s) {
        int left = 0, right = s.length() - 1;
        char[] res = s.toCharArray();
        while (left < right) {
            while (left < right && !Character.isLetter(s.charAt(left))) {
                left++;
            }
            while (left < right && !Character.isLetter(s.charAt(right))) {
                right--;
            }
            char temp = res[left];
            res[left] = res[right];
            res[right] = temp;

            left++;
            right--;
        }
        return new String(res);
    }
}
